test/e2e/nodepool_upgrade_test.go

```diff
@@ -203,4 +203,5 @@ func (ru *NodePoolUpgradeTest) Run(t *testing.T, nodePool hyperv1.NodePool, node
 		e2eutil.WithTimeout(20*time.Minute),
 	)
 	e2eutil.WaitForReadyNodesByNodePool(t, ctx, ru.hostedClusterClient, &nodePool, ru.hostedCluster.Spec.Platform.Type)
+	e2eutil.EnsureNodesRuntime(t, ctx, ru.hostedClusterClient, nodes)
```

Comment:
This would check against the original Nodes for the NodePool.

If you want to check against the ones produced by the rollout
You'll need to get them in the wait invocation e.g.
```
newNodes := e2eutil.EnsureNodesRuntime(t, ctx, ru.hostedClusterClient, nodes)
```
We can actually validate both before and after